rhino/src/test/java/org/mozilla/javascript/FunctionPrototypeSymbolHasInstanceTest.java

```diff
@@ -0,0 +1,172 @@
+        Utils.assertWithAllOptimizationLevelsES6("true:true", script);
+    }
+
+    @Test
+    public void testFunctionPrototypeSymbolHasInstanceOnObjectReturnsTrue() {
+        String script =
+                "var f = function() {};\n"
+                        + "var o = new f();\n"
+                        + "var o2 = Object.create(o);\n"
+                        + "(f[Symbol.hasInstance](o)) + ':' + "
+                        + "(f[Symbol.hasInstance](o2));\n";
+        Utils.assertWithAllOptimizationLevelsES6("true:true", script);
+    }
+
+    @Test
+    public void testFunctionPrototypeSymbolHasInstanceOnBoundTargetReturnsTrue() {
+        String script =
+                "var BC = function() {};\n"
+                        + "var bc = new BC();\n"
+                        + "var bound = BC.bind();\n"
+                        + "bound[Symbol.hasInstance](bc);\n";
+        Utils.assertWithAllOptimizationLevelsES6(true, script);
+    }
+
+    @Test
+    public void testFunctionInstanceNullVoidEtc() {
+        String script =
+                "var f = function() {};\n"
+                        + "var x;\n"
+                        + "a = (undefined instanceof f) + ':' +\n"
+                        + "(x instanceof f) + ':' +\n"
+                        + "(null instanceof f) + ':' +\n"
+                        + "(void 0 instanceof f)\n"
+                        + "a";
+        Utils.assertWithAllOptimizationLevelsES6("false:false:false:false", script);
+    }
+
+    @Test
+    public void testFunctionPrototypeSymbolHasInstanceReturnsFalseOnUndefinedOrProtoypeNotFound() {
+        String script =
+                "Function.prototype[Symbol.hasInstance].call() + ':' +"
+                        + "Function.prototype[Symbol.hasInstance].call({});";
+        Utils.assertWithAllOptimizationLevelsES6("false:false", script);
+    }
+
+    @Test
+    public void testSymbolHasInstanceIsInvokedInInstanceOf() {
+        String script =
+                ""
+                        + "var globalSet = 0;"
+                        + "var f = {\n"
+                        + "   [Symbol.hasInstance](value) { "
+                        + "       globalSet = 1;"
+                        + "       return true;"
+                        + "   }"
+                        + "}\n"
+                        + "var g = {}\n"
+                        + "Object.setPrototypeOf(g, f);\n"
+                        + "g instanceof f;"
+                        + "globalSet == 1";
+        Utils.assertWithAllOptimizationLevelsES6(true, script);
+    }
+
+    @Test
+    public void testThrowTypeErrorOnNonObjectIncludingSymbol() {
+        String script =
+                ""
+                        + "var f = function() {}; \n"
+                        + "f.prototype = Symbol(); \n"
+                        + "f[Symbol.hasInstance]({})";
+
+        Utils.runWithAllOptimizationLevels(
+                (cx) -> {
+                    cx.setLanguageVersion(Context.VERSION_ES6);
```

Comment:
there are now some assertThrows.... methods in the Utils, maybe you can use one of these

Reply:
Thanks! I've used `Utils.assertEcmaErrorES6()`